amulety/utils.py

```diff
@@ -41,49 +42,115 @@ def insert_space_every_other_except_cls(input_string: str):
+    use_cdr3_for_tcr: bool = True,
+    mode: str = "concat",
 ):
     """
-    Processes AIRR-seq data from the input file path and returns a pandas DataFrame containing the sequence to embed.
-
-    This function supports both BCR and TCR data with automatic chain mapping:
-    - BCR: IGH → H (Heavy), IGL/IGK → L (Light)
-    - TCR: TRB/TRD → H (Heavy), TRA/TRG → L (Light)
+    Processes AIRR-seq data and returns a pandas DataFrame containing sequences to embed.
 
-    It will drop cells with missing heavy or light chain if operating in single-cell only mode (no cell IDs missing) and log the number of missing chains.
-    If the data is bulk only, it will raise an error if chain = "HL".
-    If the data is mixed bulk and single-cell, and the mode is HL it will concatenate heavy and light chains per cell and drop cells with missing chains.
+    Uses AMULETY's unified H/L/HL interface for both BCR and TCR data. See embed_airr()
+    function documentation for detailed chain parameter explanations.
 
     Parameters:
         airr_df (pandas.DataFrame): Input AIRR rearrangement table as a pandas DataFrame.
-        chain (str): The input chain, which can be one of ["H", "L", "HL"].
-                    For TCR data: H=Beta chains, L=Alpha chains, HL=Alpha-Beta pairs
+        chain_mode (str): The input chain, one of ["H", "L", "HL", "LH", "H+L"].
+
         sequence_col (str): The name of the column containing the amino acid sequences to embed.
         cell_id_col (str): The name of the column containing the single-cell barcode.
         receptor_type (str): The receptor type to validate, one of ["BCR", "TCR", "all"].
                            - "BCR": validates only BCR chains (IGH, IGL, IGK) are present
                            - "TCR": validates only TCR chains (TRA, TRB, TRG, TRD) are present
                            - "all": allows both BCR and TCR chains in the same file
+        duplicate_col (str): The name of the numeric column used to select the best chain when
+                           multiple chains of the same type exist per cell. Default: "duplicate_count".
+        use_cdr3_for_tcr (bool): Whether to automatically use CDR3 sequences for TCR data when available.
+                               Default: True. Set to False to force use of sequence_col for all data.
 
     Returns:
         pandas.DataFrame: Dataframe with formatted sequences.
 
     Raises:
-        ValueError: If chain is not one of ["H", "L", "HL"] or receptor_type validation fails.
+        ValueError: If chain is not one of ["H", "L", "HL", "LH", "H+L"] or receptor_type validation fails.
     """
-    allowed_sequence_input = ["H", "L", "HL"]
-    if chain not in allowed_sequence_input:
+    allowed_sequence_input = ["H", "L", "HL", "LH", "H+L"]
+    if chain_mode not in allowed_sequence_input:
         raise ValueError(f"Input x must be one of {allowed_sequence_input}")
 
+    # Warning for LH order
+    if chain_mode == "LH":
```

Comment:
This check here is good, but maybe you can remove the other check outside this function that also checks that.